phlex/model/layer_path.cpp

```
@@ -0,0 +1,99 @@
+
+  std::size_t layer_path::hash() const noexcept
+  {
+    // incomplete paths have an implied "job" root in this calculation
+    // Need the experimental:: so it isn't confused for this function
+    std::size_t seed =
+      is_complete() ? "job"_idq.hash : experimental::hash("job"_idq.hash, layer_path_[0].hash());
+    boost::hash_range(seed, layer_path_.begin() + 1, layer_path_.end());
+    return seed;
+  }
+
+  std::set<std::size_t> layer_path::hashes() const
```

Cppcheck (2.20.0)

[style] The function 'hashes' is never used.

(unusedFunction)